File: prolog/ruleset/test_recommendations.pl
% filepath: /c:/RnD/MedHack/team54/prolog/test_recommendations.pl

% Test definition structure
% test(Name, Description, NormalRange, Priority).

% Base tests
base_test(fasting_blood_glucose, 'Measures blood sugar levels after fasting for 8 hours', '70-99 mg/dL', required).
base_test(hba1c, 'Reflects average blood glucose over past 2-3 months', 'Below 5.7%', required).

% Medium risk additional tests
medium_risk_test(ogtt, 'Measures how well your body processes glucose', 'Below 140 mg/dL after 2 hours', required).
medium_risk_test(fasting_insulin, 'Measures insulin resistance', '3-25 mIU/L', recommended).
medium_risk_test(lipid_profile, 'Measures different types of cholesterol', 'Varies by component', recommended).

% High risk additional tests
high_risk_test(c_peptide, 'Measures insulin production by pancreas', '0.5-2.0 ng/mL', required).
high_risk_test(gad_antibodies, 'Helps distinguish between type 1 and type 2 diabetes', 'Below 5 IU/mL', recommended).
high_risk_test(microalbuminuria, 'Checks for kidney damage', 'Below 30 mg/24 hours', required).

% Rules to get recommendations based on probability
get_test_recommendations(Probability, Category, Tests) :-
    Probability < 0.3,
    Category = 'LOW_RISK_TESTS',
    findall([Name, Desc, Range, Priority], 
            base_test(Name, Desc, Range, Priority), 
            Tests).

get_test_recommendations(Probability, Category, Tests) :-
    Probability >= 0.3,
    Probability < 0.7,
    Category = 'MEDIUM_RISK_TESTS',
    findall([Name, Desc, Range, Priority],
            (base_test(Name, Desc, Range, Priority);
             medium_risk_test(Name, Desc, Range, Priority)),
            Tests).

get_test_recommendations(Probability, Category, Tests) :-
    Probability >= 0.7,
    Category = 'HIGH_RISK_TESTS',
    findall([Name, Desc, Range, Priority],
            (base_test(Name, Desc, Range, Priority);
             medium_risk_test(Name, Desc, Range, Priority);
             high_risk_test(Name, Desc, Range, Priority)),
            Tests).

File: app/src/utils/testRecommendations.ts
export interface TestRecommendation {
  name: string;
  description: string;
  normalRange: string;
  priority: 'required' | 'recommended' | 'optional';
}

export interface TestResponse {
  category: string;
  tests: TestRecommendation[];
}

const baseTests: TestRecommendation[] = [
  {
    name: "Fasting Blood Glucose (FBG)",
    description: "Measures blood sugar levels after fasting for 8 hours",
    normalRange: "70-99 mg/dL",
    priority: "required"
  },
  {
    name: "HbA1c",
    description: "Reflects average blood glucose over past 2-3 months",
    normalRange: "Below 5.7%",
    priority: "required"
  }
];

const mediumRiskTests: TestRecommendation[] = [
  ...baseTests,
  {
    name: "Oral Glucose Tolerance Test (OGTT)",
    description: "Measures how well your body processes glucose",
    normalRange: "Below 140 mg/dL after 2 hours",
    priority: "required"
  },
  {
    name: "Fasting Insulin Level",
    description: "Measures insulin resistance",
    normalRange: "3-25 mIU/L",
    priority: "recommended"
  },
  {
    name: "Lipid Profile",
    description: "Measures different types of cholesterol",
    normalRange: "Varies by component",
    priority: "recommended"
  }
];

const highRiskTests: TestRecommendation[] = [
  ...mediumRiskTests,
  {
    name: "C-Peptide Test",
    description: "Measures insulin production by pancreas",
    normalRange: "0.5-2.0 ng/mL",
    priority: "required"
  },
  {
    name: "GAD Antibodies",
    description: "Helps distinguish between type 1 and type 2 diabetes",
    normalRange: "Below 5 IU/mL",
    priority: "recommended"
  },
  {
    name: "Microalbuminuria",
    description: "Checks for kidney damage",
    normalRange: "Below 30 mg/24 hours",
    priority: "required"
  }
];

export function getTestRecommendations(probability: number): TestResponse {
  if (probability < 0.3) {
    return {
      category: "LOW_RISK_TESTS",
      tests: baseTests
    };
  }

  if (probability < 0.7) {
    return {
      category: "MEDIUM_RISK_TESTS",
      tests: mediumRiskTests
    };
  }

  return {
    category: "HIGH_RISK_TESTS",
    tests: highRiskTests
  };
}